src/client/Microsoft.Identity.Client/ManagedIdentity/AzureArcManagedIdentitySource.cs

```diff
@@ -29,7 +29,16 @@ internal class AzureArcManagedIdentitySource : AbstractManagedIdentity
 
         public static AbstractManagedIdentity Create(RequestContext requestContext)
         {
-            string identityEndpoint = EnvironmentVariables.IdentityEndpoint;
+            string identityEndpoint;
+                
+            if (EnvironmentVariables.IdentityEndpoint == null)
```

Comment:
When `AzureArcManagedIdentitySource` gets executed, it will always return true because the default endpoint is set and will never execute IMDS?